docs(cn): add react without es6 translate (#128)
reactjs/zh-hans.reactjs.org: content/docs/react-without-es6.md

});
```

ES6 中的 class 与 `createReactClass` 方法十分相似，但有以下几个区别值得注意。

## 声明默认属性 {#declaring-default-props}

> 如果使用函数和 ES6 的 class 关键字创建组件，可以直接把自定义属性对象写到组件的 `defaultProps` 属性中：

Author: 译者注什么？

> 如果使用函数和 ES6 的 class 关键字创建组件，可以直接把自定义属性对象写到组件的 `defaultProps` 属性中：

Reviewer: 这块和 @QC-L 确认一下吧。译者注有标准表达的话就统一，没有的话在译文后方放一个`（译者注）`就可以吧。

> 如果使用函数和 ES6 的 class 关键字创建组件，可以直接把自定义属性对象写到组件的 `defaultProps` 属性中：

Reviewer: @zhaozhiming 在补充语义的地方采用后者方式进行标注吧，如为链接可以加个脚标。

> 如果使用函数和 ES6 的 class 关键字创建组件，可以直接把自定义属性对象写到组件的 `defaultProps` 属性中：

Reviewer 2: ```suggestion
无论是函数组件还是 class 组件，都拥有 `defaultProps` 属性：
```

};
```

> 如果使用 `createReactClass` 方法创建组件，那就需要在参数对象中定义 `getDefaultProps` 方法，并且在这个方法中返回包含自定义属性的对象：

Reviewer 3: 不用改删掉括号吧。

});
```

> ## 设置初始状态 {#setting-the-initial-state}

Reviewer 2: ```suggestion
## 初始化 State {#setting-the-initial-state}
```

});
```

> 这就意味着，如果使用 ES6 class 关键字创建组件，那在处理事件回调的时候就要多写一点点代码。但对于大型项目来说，这样做可以提升运行效率。

Reviewer 2: ```suggestion
这就意味着，如果使用 ES6 class 关键字创建组件，在处理事件回调的时候就要多写一部分代码。但对于大型项目来说，这样做可以提升运行效率。
```

> 如果你觉得上面这个写法很麻烦，那么可以尝试一下 **目前还处于试验性阶段** 的 Babel 插件 [Class Properties](https://babeljs.io/docs/plugins/transform-class-properties/)。

Reviewer 2: ```suggestion
如果你觉得上面这个写法很麻烦，那么可以尝试一下**目前还处于试验性阶段**的 Babel 插件 [Class Properties](https://babeljs.io/docs/plugins/transform-class-properties/)。
```

}
```

请注意，上面这种语法**目前还处于试验性阶段**，这意味着语法随时都可能改变，也存在最终不被列入框架标准的可能。

为了保险起见，以下三种做法都是可以的：

* 在 constructor 中绑定方法。
* 使用箭头函数，比如：`onClick={(e) => this.handleClick(e)}`。
* 继续使用 `createReactClass`。

## Mixins {#mixins}

> >**注：**

Reviewer 2: ```suggestion
>**注意：**
```

>

> >ES6 本身是不包含 mixin 支持的。因此，如果你使用 ES6 class 关键字创建组件，那就不能使用 mixin 功能了。

Reviewer: 1. 「不包含任何 mixins 支持」
2. 「因此，当你在 React 中使用 ES6 class 时，将不支持 mixins 。」

>

> >**我们也发现了很多使用混入然后出现了问题的代码库。[因此，我们并不推荐在 ES6 中使用 mixin ](/blog/2016/07/13/mixins-considered-harmful.html)。**

Reviewer: mixins 不翻译，前后文保持一致。

> >**我们也发现了很多使用 mixin 然后出现了问题的代码库。[因此，我们并不推荐在 ES6 中使用 mixin ](/blog/2016/07/13/mixins-considered-harmful.html)。**

Reviewer 3: [并且不建议在新代码中使用 mixin ]

>
> 以下内容仅作为参考。

> 如果完全不同的组件有相似的功能，这就会产生 ["横切关注点" 问题](https://en.wikipedia.org/wiki/Cross-cutting_concern)。针对这个问题，在使用 createReactClass 创建 React 组件的时候，引入 `mixin` 功能会是一个很好的解决方案。

Reviewer: 1. 空格问题
2. mixins 不译且保持原文，s 不需删除

Author: mixins 是英文中是复数，但在中文语境感觉不需要用到复数，是不是始终用单数形式 `mixin` 比较好？

Reviewer: 有较多地方也在使用 mixins 例如 Vue.js 文档，mixin 和 mixins 在不少地方都同指一个概念，既然不译就暂时保持原文好了。

> 如果完全不同的组件有相似的功能，这就会产生["横切关注点" 问题](https://en.wikipedia.org/wiki/Cross-cutting_concern)。针对这个问题，在使用 createReactClass 创建 React 组件的时候，引入 `mixin` 功能会是一个很好的解决方案。

Reviewer 3: 把英文也加上 横切关注点 是个专业名词，英文反而更好理解

Author: 这样吗？`"横切关注点" 问题（cross-cutting concerns）`

Reviewer 2: "横切关注点（cross-cutting concerns）"

> 一个常见的使用情景是，当一个组件想要每隔一段时间更新，那么最简单的方法就是使用 `setInterval()`。但更重要的是，如果后续代码中不需要这个功能，为了节省内存，你应该把它删除。React 提供了 [生命周期方法](/docs/working-with-the-browser.html#component-lifecycle)，这样你就可以知道某一个组件什么时候要被创建或被销毁。我们先来创建一个使用 `setInterval()` 的 mixin，它会在组件销毁的时候也销毁。

Reviewer: 「一个常见的使用情景是，一个组件每隔一段时间需要更新。使用 `setInterval()` 可以很容易实现这个功能，但更重要的是，当你不再需要它时，为了节省内存，你应该把它删除。」
「这样你就可以知道一个组件何时被创建或被销毁了。」
「让我们创建一个简单的 mixin，它使用这些方法提供一个简单的 `setInterval()` 函数，它会在你的组件被销毁时被自动清理。」

> 一个常见的使用情景是，一个组件每隔一段时间需要更新。使用 `setInterval()` 可以很容易实现这个功能，但更重要的是，当你不再需要它时，为了节省内存，你应该把它删除。React 提供了[生命周期方法](/docs/working-with-the-browser.html#component-lifecycle)，这样你就可以知道一个组件何时被创建或被销毁了。让我们创建一个简单的 mixin，它使用这些方法提供一个简单的 `setInterval()` 函数，它会在组件被销毁时被自动清理。

Reviewer 3: 当你不再需要它时，你应该删掉它已节省内存。

> 一个常见的使用情景是，一个组件每隔一段时间需要更新。使用 `setInterval()` 可以很容易实现这个功能，但更重要的是，当你不再需要它时，你应该删掉它以节省内存。React 提供了[生命周期方法](/docs/working-with-the-browser.html#component-lifecycle)，这样你就可以知道一个组件何时被创建或被销毁了。让我们创建一个简单的 mixin，它使用这些方法提供一个简单的 `setInterval()` 函数，它会在组件被销毁时被自动清理。

Reviewer 2: ```suggestion
比较常见的用法是，组件每隔一段时间更新一次。使用 `setInterval()` 可以很容易实现这个功能，但需要注意的是，当你不再需要它时，你应该清除它以节省内存。React 提供了[生命周期方法](/docs/working-with-the-browser.html#component-lifecycle)，这样你就可以知道一个组件何时被创建或被销毁了。让我们创建一个简单的 mixin，它使用这些方法提供一个简单的 `setInterval()` 函数，它会在组件被销毁时被自动清理。
```

);
```

> 如果一个组件有多个 mixin，且其中几个 mixin 中定义了相同的生命周期方法（比如都会在组件被摧毁的时候执行），那么这些生命周期方法是一定会被调用的。通过 mixin 定义的方法，执行顺序也与定义时的顺序一致，且会在组件上的方法执行之后再执行。

Reviewer: 如果一个组件使用多个 mixins，且几个 mixins 定义了相同的生命周期方法（例如，当组件被销毁时，几个 mixins 都想要进行一些清理工作），所有生命周期方法都确保会被调用。mixins 上定义的方法会按照 mixins 被列出的顺序依次执行，紧接着是在组件上进行方法调用。

@QC-L 确认下后半句的语义。

Reviewer 2: ```suggestion
如果组件拥有多个 mixin，且这些 mixin 中定义了相同的生命周期方法（比如都会在组件被摧毁的时候执行），那么这些生命周期方法都会被调用的。使用 mixins 时，mixins 会先按照定义时的顺序执行，最后调用组件上对应的方法。
```
